lib/google/amp/cache/invalidator.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+require 'net/http'
+require_relative './signatory'
+require_relative './errors'
+
+module Google
+  module AMP
+    module Cache
+      class Invalidator
+        CONTENT_TYPE_MAPPING = {
+          document: :c,
+          image: :i,
+          resource: :r
+        }.freeze
+
+        PROTOCOL_MAPPING = {
+          https: :s,
+          http: nil
+        }.freeze
+
+        def initialize(url, content_type: :document)
+          @url = url
+          @content_type = CONTENT_TYPE_MAPPING.fetch(content_type.to_sym, :c)
+        end
+
+        def ping
```

Comment:
While looking at this method, I thought about the deprecated `update-ping` method from AMP cache. Could we change this method naming?

Reply:
Now I see the ambiguity naming issue, this should definitely be changed

Is `#update_cache` fine? or do you have any other preference?

Reply:
@dlackty I've updated accordingly, please let me know if there's anything else missing